internal/utils/miscellaneous_test.go

```diff
@@ -0,0 +1,255 @@
+package utils
+
+import (
+	"testing"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestMergeMaps(t *testing.T) {
```

Comment:
i already added unit tests for existing methods in the API module PR. Since you have added this in, i will perhaps remove what i have added then. Lets sync up before investing time so that duplicate efforts can be avoided in future.

Reply:
Sure. I wasn't aware of that, and since I saw missing unit tests, I simply added them here.